일기 상세 페이지 > 댓글 영역 UI 구현

Bori-github opened the pull request:
## 🔗 연관된 이슈

- close #48 

<br />

## 🗒 작업 목록

- [x] Comment list 목 데이터 생성
  - 목 데이터 이미지 도메인 next.config에 추가
- [x] 댓글 영역 아이콘 svg 추가
- [x] 댓글 목록 UI 구현
  - 댓글 있는 경우/없는 경우 
- [x]  댓글 입력 UI 구현
- [x] 댓글 입력 textarea 자동 높이 조절 기능 구현
- [x] 댓글 쓰기 버튼 UI 구현
  - label의 htmlFor 기능 이용하여 input 포커싱 구현
- [x] 댓글 입력 시 send 버튼 활성화 UI 구현
- [x] 댓글 영역 추상화
  - DiaryContainer, DiaryComments, DiaryComment, DiaryCommentInput 컴포넌트 분리
  - 네이밍 컨벤션에 맞춰 수정

<br />

## 🧐 PR Point

- 댓글 목 데이터 추가하면서 next.config에 목 데이터 이미지 도메인 추가
⇒ 추후 목 데이터 삭제 시 이미지 도메인 삭제 예정
- 댓글의 신고하기/삭제/수정 모달은 게시글 모달과 함께 적용할 예정입니다.
- 네이밍 적절한지 확인 부탁드립니다.
- 컴포넌트 분리를 하나의 커밋으로 한꺼번에 진행했습니다. 이럴 경우 나눠서 하는게 더 좋을지 한꺼번에 해도 좋을지 이야기 나눠보면 좋을 거 같습니다.
- sr-only 처리가 필요한 부분이 생겼으나 다른 브랜치에서 sr-only 스타일 생성  후 적용할 예정입니다.
- #46 머지 이후 해당 브랜치 머지 진행할 예정입니다.

<br />

## 💥 Trouble Shooting
- 해당 작업을 하던 중 발생했던 문제에 대해 작성해주세요.

<br />

## 📸 스크린샷 / 피그마 링크
<img width="286" alt="image" src="https://user-images.githubusercontent.com/85009583/226083544-6cdf088d-4350-49a4-8be8-df56c85ea55c.png">

<br />

## 📚 참고

- 컴포넌트 분리하면서 폴더구조, 파일명, 컴포넌트명 참고한 레포지토리 : [벨로그 레포지토리 frontend 디렉토리](https://github.com/velopert/velog/tree/master/velog-frontend)

<br />

## ✅ PR Submit 전 체크리스트

- [x] Merge 하는 브랜치는 `main` 브랜치가 아닙니다. 
- [x] 코드에 크리티컬한 `error` 또는 `warning`이 존재하지 않습니다.
- [x] 불필요한 `console`이 존재하지 않습니다.

Bori-github committed:
🤡 #48 - Comment list 목데이터 생성

Bori-github committed:
🔧 #48 - 목 데이터 이미지 domain 추가

Bori-github committed:
🍱 #48 - 댓글 영역 아이콘 svg 추가

- more, reply, send_active, send_inactive 추가

Bori-github committed:
💄 #48 - 댓글 목록 UI 구현

- 댓글이 없는 경우 UI
- 댓글 목록 UI

Bori-github committed:
💄 #48 - 댓글 입력 영역 UI 구현

Bori-github committed:
✨ #48 - 댓글 입력 textarea 자동 높이 조절 기능 구현

Bori-github committed:
💄 #48 - 댓글쓰기 버튼 UI 구현

- label을 이용하여 클릭 시 댓글 입력 창으로 포커스 이동 구현

Bori-github committed:
💄 #48 - 댓글 입력 시 send 버튼 활성화 UI 구현

yws1502 commented on src/pages/diary/[id].tsx:
```diff
@@ -156,3 +166,46 @@ const NoCommentText = styled.p`
+  background-color: ${({ theme }) => theme.colors.white};
+`;
+
+const ReplyBox = styled.div`
```
ReplyBox의 태그를 div로 설정해주셨는데, form 태그를 사용하지 않으신 이유가 있으신가요?!

form 태그를 사용하면 textarea에서 엔터키가 줄바꿈 대신 submit 이벤트로 작동하는 것으로 알고 있습니다!
댓글 영역에서의 엔터키가 줄바꿈이랑 submit 중에 어떤 것이 사용자에게 편할지 이야기 나눠보면 좋을 것 같습니다~!

Bori-github replied:
textarea에서 엔터 가능 여부에 대해서 고민을 하다가 일단 이 상태를 유지했습니다.
그리고 댓글에서도 엔터가 적용된 상태로 보여지는게 좋은지도 같이 이야기 해보면 좋을 거 같아요.

회의 때 해당 내용들 결정하고 이어서 진행하겠습니다!

yws1502 commented on src/containers/diary/DiaryCommentsContainer.tsx:
```diff
@@ -0,0 +1,62 @@
+  modifiedAt: string;
+}
+
+const DiaryCommentsContainer = () => {
```
containers라는 폴더를 별도로 분리하신 이유가 궁금합니다!

components/diary의 diary component와 비슷한 구조로 diary 폴더에 comment depth를 한단계 더 늘리면 폴더 depth가 너무 깊어질까요??
예를 들어 components/diary/comment 경로에 comment관련된 컴포넌트들을 모아두는 식으로요! 
(commentContainer, commentList, comment, commentInput)

Bori-github replied:
위의 참고 링크에 벨로그 레포지토리 참고해서 폴더 구조를 잡아보았습니다.

#### 참고 레포지토리의 폴더 구조 특징
1. 같은 레벨의 components와 containers 폴더
2. containers를 폴더로 분리하여 관리
3. 게시글과 댓글 관련 컴포넌트를 같은 폴더에서 관리

#### 1. 폴더 레벨
- 컴포넌트가 컨테이너보다 작은 단위라고 생각한다면 각 폴더가 같은 레벨인 게 적절하다고 생각합니다.
⇒ 저는 이렇게 생각해서 같은 레벨에 두 폴더를 위치했습니다.
- 컨테이너도 컴포넌트라고 할 수 있다라고 생각한다면 컴포넌트 하위 레벨에 컨테이너가 적절하다고 생각합니다.

#### 2. containers를 폴더
- 컴포넌트를 따로 분리해서 관리하는 것처럼 컨테이너도 따로 분리해서 관리하면 관리에 용이하지 않을까 해서 적용했습니다.
- 실제로 참고했던 레포지토리의 폴더 구조에서 코드를 찾아볼 때 구조가 이해 되면서 코드 찾아보기 수월했습니다.

#### 3. 댓글 관련 컴포넌트 관리
- diary 폴더와 같은 레벨에 comment 폴더 생성하기
⇒ 현재 댓글은 일기 게시글에만 있으므로 같은 레벨에 있는 것은 적절하지 않다고 생각했습니다.
추후에 일기 게시글이 아닌 다른 게시글을 작성할 수 있는 기능이 구현된다면 이 방법을 사용하는 것이 좋다고 생각합니다.
- comment 폴더를 diary 폴더 내에 생성하기
⇒ depth가 깊어지는 것은 최대한 피하고 싶어서 diary 내에 comment 폴더를 생성하지 않았습니다.
참고 레포지토리에 게시글과 댓글 관련 컴포넌트가 같은 폴더 내에서 관리되고 있는 것을 확인하고 같은 방식을 적용했습니다.

폴더 구조를 설계하면서 여러 레포지토리를 참고하지 않고, 특정 레포지토리의 방법을 그대로 사용했습니다.
관련 내용 조금 더 찾아보고 회의 때 내용 공유하겠습니다.

yws1502 replied:
순전히 container라는 폴더를 분리하신 이유가 궁금해서 댓글 달았었습니다..! 그런데 참고 링크를 확인하지 못했네요 ㅜㅜ

container 폴더를 분리하면 확실히 컴포넌트 흐름 찾아가기고 용이하겠네요 :) 상세하게 답글 달아주셔서 감사합니다!!

yws1502 reviewed (commented):
추상화할 컴포넌트의 양이 많지 않아서 그런지 모르겠지만, 이정도 양이라면 이렇게 하나의 커밋으로 해주셔서 확인하기 편했던 것 같습니다~! 추상화도 의미에 맞게 잘된 것 같아요. 고생하셨습니다 :)

Bori-github committed:
♻️ #48 - 댓글 영역 추상화

- DiaryContainer, DiaryComments, DiaryComment, DiaryCommentInput 컴포넌트 분리
- 네이밍 컨벤션에 맞춰 수정

Bori-github committed:
⚡️ #48 - CommentTextarea에 react hook form 적용

- CommentTextarea의 부모 태그 div > form으로 변경
- CommentTextarea 최대 높이 적용
- useRef 삭제

Bori-github committed:
💄 #48 - textarea 값인 댓글 개행 처리

Bori-github committed:
💄 #48 - CommentTitle에 screen reader only 스타일 적용

Bori-github committed:
✨ #48 - 댓글 글자수 제한 기능 구현

- 댓글 글자 수 확인하여 최대 글자 수를 넘길 경우 최대 글자 수까지 자르고, alert로 에러 메시지를 보여준다.
- textarea 자동 높이 조절 함수 분리하여 적용
